# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/webrender_bindings/RenderCompositorANGLE.h
# Commit: edfb73b96d5a
# Full Hash: edfb73b96d5ad946bbdd9da2e98d07b1f4565e1f
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2019-06-15 21:44:24
# Regressor Bug: 1559524
# File Overlap Count: 2
# Description:
#   Bug 1559524 - Reduce calls to GLLibraryEGL::Get(). r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D35103
# ==============================================================================

diff -r 0378898329b2 -r edfb73b96d5a gfx/webrender_bindings/RenderCompositorANGLE.h
--- a/gfx/webrender_bindings/RenderCompositorANGLE.h	Sat Jun 15 05:14:53 2019 +0000
+++ b/gfx/webrender_bindings/RenderCompositorANGLE.h	Sat Jun 15 05:21:12 2019 +0000
@@ -24,6 +24,9 @@
 struct IDXGISwapChain;
 
 namespace mozilla {
+namespace gl {
+class GLLibraryEGL;
+}  // namespace gl
 
 namespace wr {
 
@@ -32,7 +35,8 @@
   static UniquePtr<RenderCompositor> Create(
       RefPtr<widget::CompositorWidget>&& aWidget);
 
-  explicit RenderCompositorANGLE(RefPtr<widget::CompositorWidget>&& aWidget);
+  explicit RenderCompositorANGLE(RefPtr<widget::CompositorWidget>&& aWidget,
+                                 gl::GLLibraryEGL*);
   virtual ~RenderCompositorANGLE();
   bool Initialize();
 
@@ -64,6 +68,7 @@
   bool SutdownEGLLibraryIfNecessary();
   RefPtr<ID3D11Query> GetD3D11Query();
 
+  RefPtr<gl::GLLibraryEGL> mEgl;
   EGLConfig mEGLConfig;
   EGLSurface mEGLSurface;
 